guardian/plutohelperagent
# Open Paths Directly from URL

## What does this change?

Open paths directly from the URL rather than querying pluto-core with a project id. to get the correct path.

The paths are checked against two white lists before an attempt is made at opening the file. One white list is for the start of the path and the other the file extension.

A white list is also now used to check the start of asset folder paths before attempting to open them.

The user can edit the white lists in the Preferences window.

Please note, this should not be merged before the required changes to pluto-core have been made as it will render the software useless without them. These changes can be found in the merge request at the following URL: -

https://gitlab.com/codmill/customer-projects/guardian/pluto-core/-/merge_requests/150

## How to test

Install the required changes to pluto-core on a server. Run the new build of PlutoHelperAgent. Add the required paths and extensions for the file you want to try and open. Press the open button in pluto-core. The file should be passed to the open script you have set up. If not, you should see an alert window explaining what went wrong.

## How can we measure success?

Project files can now be opened directly from the URL which does not require the user to be logged into Pluto in PlutoHelperAgent.

## Have we considered potential risks?

Yes. The white lists should stop most if not all attempts at running malicious software.

## Images

![image](https://user-images.githubusercontent.com/10620802/154484154-6cb3ac67-8768-4e04-8103-b66cab5ba55c.png)
